client/state/breadcrumb/reducer.js

```diff
@@ -0,0 +1,34 @@
+import {
+	BREADCRUMB_RESET_LIST,
+	BREADCRUMB_UPDATE_LIST,
+	BREADCRUMB_APPEND_ITEM,
+} from 'calypso/state/action-types';
+import { combineReducers, withSchemaValidation } from 'calypso/state/utils';
+import { breadcrumbSchema } from './schema';
+
+// Stores the complete list of breadcrumbs in an array,
+export const items = withSchemaValidation( breadcrumbSchema, ( state = [], action ) => {
+	switch ( action.type ) {
+		case BREADCRUMB_RESET_LIST:
+			return [];
+		case BREADCRUMB_UPDATE_LIST:
+			return [ ...action.items ];
+		case BREADCRUMB_APPEND_ITEM:
+			// Don't append if it is the same as the last item
+			// eslint-disable-next-line no-case-declarations
+			const currentLastItem = state.at( -1 ) || {};
+			if ( currentLastItem.label === action.item?.label ) {
+				return [ ...state ];
+			}
+
+			return [ ...state, action.item ];
+	}
+
+	return state;
+} );
+
+const combinedReducer = combineReducers( {
```

Comment:
It seems like `combineReducers` is not needed since we are exporting only one reducer?

Reply:
Yeah, added as a "for the future" thing, but I think we can remove it for now.

Removed at https://github.com/Automattic/wp-calypso/pull/60491/commits/1814a84396098f3d617714d02c85292340a9de56